.github/workflows/build-and-test-other.yaml

```diff
@@ -25,33 +25,81 @@ env:
   elixir_version: 1.14
 
 jobs:
+  compile_tests:
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+
+    - uses: erlef/setup-beam@v1
+      with:
+        otp-version: ${{ env.otp_version }}
+        elixir-version: ${{ env.elixir_version }}
+
+    - name: apt update
+      run: sudo apt update
+
+    - name: Install required packages
+      run: sudo apt install -y gperf
+
+    - name: Compile test modules
+      run: |
+        set -e
+        mkdir build_tests
+        cd build_tests
+        cmake ..
+        make erlang_test_modules
+        make test_estdlib
+        make test_eavmlib
+        make test_alisp
+
+    - name: Upload test modules
+      uses: actions/upload-artifact@v3
```

Comment:
For my understadning, usually we use cache action for this kind of stuff, did we find any shortcoming in that action?

Reply:
Cache and artifacts passed between jobs serve very different purposes. We could use both. I wanted to only compile beam files once for all tested architectures. A cache would mean to try to not compile them for subsequent runs of the workflow.

Reply:
what worries me is that if we do this in this way we might get wrong results as soon as we make any change to the .erl files.
so we need to build an unique key using `hashFiles` and the otp version (and stuff like that) in order to invalidate the artifacts when we change something.

Please, correct me if I am wrong here.

Reply:
This logic is not present yet in the project but is fairly common.

The workflow is divided in two jobs. Everytime it is run, both jobs are executed. The first job compiles the beam and uploads an artifact that the second job picks up.

We could delete the artifact afterwards but the retention policy is set to a minimum of 1 day so I didn't bother. Any new run of the workflow will produce a new artifact and the uploaded artifact is only used here.